### PR TITLE
[fix] Make scale dependent visibility more robust for labels

### LINKED ISSUE #42443: Label visibility by scale max/min values inconsitencies
I'm testing Qgis 3.16.5, but it is also in Qgis 3.10.8 (actually used).

I'm a little bit confused and I can't understand if it's a real bug or a misunderstandings of mine, but If I create a label and I check the "scale intervals" option I have to select "minimum (excluded) and "maximum (included)".
So I set "minimum" at 1.000.000 and "maximum" at 500.000
From what I understand at a scale between "min" and "max" and at "max" I should see the label, while I shouldn't see it at "minimum".
A part the fact I can't understand the logic to exclude one of the two extremes, actually if I set scale 1:1.000.000 I see the label that I shouldn't and at 1:500.000 scale I don't see it while I should. So basically Qgis does the opposite.
This happens in the CANVAS.
If I create a Layout and I draw 2 windows, one at 1:1.000.000 and the other at 1:500.000, I have the label in both of them!

### PR DESCRIPTION
**Pull request by gacarrillor**
Followup https://github.com/qgis/QGIS/pull/58968

Make labeling more robust when handling scale dependent visibility: handle exclusive/inclusive behavior on edge cases (non-round numbers).

Also, add tooltip for scale dependent visibility widgets (labeling) to indicate that min scale is exclusive and max scale is inclusive.

image

------

Fix https://github.com/qgis/QGIS/issues/42443

### CODE REVIEW

File: src/ui/qgstextformatwidgetbase.ui [hunk 1]
```diff
@@ -35,10 +35,10 @@
       </size>
      </property>
      <property name="frameShape">
-      <enum>QFrame::NoFrame</enum>
+      <enum>QFrame::Shape::NoFrame</enum>
```
Comment (nyalldawson) [inline src/ui/qgstextformatwidgetbase.ui:38]: These changes can break things on qt 5 builds -- can you revert them?

File: src/ui/qgstextformatwidgetbase.ui [hunk 2]
```diff
@@ -2732,7 +2732,7 @@ font-style: italic;</string>
                               <string>Size</string>
                              </property>
                              <property name="alignment">
-                              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+                              <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
```
Comment (nirvn) [inline src/ui/qgstextformatwidgetbase.ui:2735]: Editing .ui files with your favorite text editor is often much easier and less disruptive than via QtCreator :)

**Review by nirvn — COMMENTED** [review]
Good to merge provided the .ui changes are reverted to only include the tooltip string changes.

File: src/ui/qgstextformatwidgetbase.ui (hunk 1 above)
Comment (gacarrillor) [reply src/ui/qgstextformatwidgetbase.ui]: Thanks for letting me know.
I just replaced the commit on the `.ui` file avoiding all those edits.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**